src/sprightly.ts

```diff
@@ -1,52 +1,43 @@
-import { readFileSync } from "fs"
+import { readFileSync, existsSync } from "fs"
 import path from "path"
 
 interface Data {
   [key: string]: string | number | Array<string | number | Data> | Data
 }
 
-class KeyNotFoundError extends Error {
-  readonly key: string
-  readonly src: string
-  constructor(key: string, src: string) {
-    const message = `Key "${key}" not found in "data" at ${src}`
-    super(message)
-    this.key = key
-    this.src = src
-  }
-}
-
-class PathNotFoundError extends Error {
-  readonly component: string
-  readonly src: string
-  constructor(component: string, src: string) {
-    const message = `Component "${component}" was not found in ${src}`
-    super(message)
-    this.component = component
-    this.src = src
-  }
-}
-
 interface Options {
   keyFallback?: string
   throwOnKeyNotfound?: boolean
   cache?: boolean
 }
 
-function parse(file: string, data: Data, options: Options) {
+function parse(file: string, entryPoint: string, data: Data, options: Options) {
```

Comment:
I think `entryPoint` should be `filename`, and `file` should become `fileContent`.

Reply:
I changed `file` to `fileContent`, but `entryPoint` is a path to a file, this is why I called it `entryPoint`. I changed it to filePath tho

Reply:
Aha, nice.